src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -624,7 +624,7 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
                 returnType = inferredReturnType.TypeWithAnnotations;
                 returnRefKind = inferredReturnType.RefKind;
 
-                if (!returnType.HasType && returnTypes.Count > 0)
+                if (!returnType.HasType && inferredReturnType.NumExpressions > 0)
```

Comment:
>if (!returnType.HasType && inferredReturnType.NumExpressions > 0) [](http://example.com/codeflow?start=16&length=65)

It looks like the change only affects scenarios that involve return statements without expressions. Is this correct? If so, consider adjusting commit's title/description as right now it feels too broad.